### ISSUE TITLE
key repeat is intermittent on tulip hardware

### ISSUE THREAD

**Issue by bwhitman** [bwhitman/tulipcc#94]
### Discussed in https://github.com/bwhitman/tulipcc/discussions/83

<div type='discussions-op-text'>

<sup>Originally posted by **drepetto** May 19, 2023</sup>
Key repeat sometimes works in terminal, almost never works in edit.py. When it does work it's just for a bit then it konks out.  I'm using an old Dell USB keyboard, no hub or anything fancy about it.</div>

**Comment by bwhitman** [comment]
Three problems here:

(1) we have big voltage drop on the r6 board from USBC in `5V_USB` over to the USB_KB 5V output. It's bad enough that the voltage to the serial chip (CH340) is going so low it's heating up. To make this more reliable, i had to bodge from the `3V3` output of the regulator to pin 4 of the CH340, and lift pin 16 of the CH340. (DAn points out this should actually be 3V3 to pin 16, and leave pin 4 lifted. But my way seems to be fine.)

(2) this keyboard (and maybe many others?) outputs a packet every few hundred ms no matter if a key is held or not. Maybe this is standard, and the 5 I've tried don't do it -- or maybe we stopped doing this on "modern" keyboards. Either way, we should handle it. So we've updated `usb_keyboard.c` to accept new packets with old information and not reset the key held counter

(3) We were calling `usbh_setup` twice in `main.c`. Unsurprisingly, the system was sensitive to this at bootup. We've fixed that.
